pkg/detectors/endpoint_customizer_test.go

```diff
@@ -9,8 +9,28 @@ import (
 func TestEmbeddedEndpointSetter(t *testing.T) {
 	type Scanner struct{ EndpointSetter }
 	var s Scanner
+	// set useFoundEndpoints to true to add the "baz" in endpoints
```

Comment:
suggestion/question: Would it make sense to split this into subtests or separate tests? The EndpointSetter has always been a bit confusing for me, so clearer labels on the tests might make them easier to grok.

Ex:
```
func TestEmbeddedEndpointSetter(t *testing.T) {
    t.Run("useFoundEndpoints is true", func(t *testing.T) {
        var s Scanner
        s.useFoundEndpoints = true

        // "baz" is passed to Endpoints, should appear in the result
        assert.Equal(t, []string{"baz"}, s.Endpoints("baz"))
    })

    t.Run("setting configured endpoints", func(t *testing.T) {
        var s Scanner

        // Setting "foo" and "bar"
        assert.NoError(t, s.SetConfiguredEndpoints("foo", "bar"))

        // Returning error because no endpoints are passed
        assert.Error(t, s.SetConfiguredEndpoints())
    })

    t.Run("useFoundEndpoints adds new endpoints", func(t *testing.T) {
        var s Scanner
        s.useFoundEndpoints = true

        // "foo" and "bar" are configured
        assert.NoError(t, s.SetConfiguredEndpoints("foo", "bar"))
        // "baz" is added because useFoundEndpoints is true
        assert.Equal(t, []string{"foo", "bar", "baz"}, s.Endpoints("baz"))
    })

    t.Run("useCloudEndpoint is true", func(t *testing.T) {
        var s Scanner
        s.useCloudEndpoint = true
        s.cloudEndpoint = "test"

        // "foo" and "bar" must still be configured
        assert.NoError(t, s.SetConfiguredEndpoints("foo", "bar"))
        // "test" is added because useCloudEndpoint is true and cloudEndpoint is set
        assert.Equal(t, []string{"foo", "bar", "test"}, s.Endpoints())
    })

    t.Run("disable both foundEndpoints and cloudEndpoint", func(t *testing.T) {
        var s Scanner
        s.useFoundEndpoints = true
        s.useCloudEndpoint = true
        s.cloudEndpoint = "test"
        assert.NoError(t, s.SetConfiguredEndpoints("foo", "bar"))

        // Confirm "test" is in the list
        assert.Equal(t, []string{"foo", "bar", "test"}, s.Endpoints())

        // Now disable both
        s.useFoundEndpoints = false
        s.useCloudEndpoint = false

        // "test" won't be added
        assert.Equal(t, []string{"foo", "bar"}, s.Endpoints("test"))
    })

    t.Run("cloudEndpoint not added when useCloudEndpoint is false", func(t *testing.T) {
        var s Scanner
        s.useFoundEndpoints = false
        s.useCloudEndpoint = false
        s.cloudEndpoint = "new"

        // "foo" and "bar" configured
        assert.NoError(t, s.SetConfiguredEndpoints("foo", "bar"))
        // "new" is not added because useCloudEndpoint is false
        assert.Equal(t, []string{"foo", "bar"}, s.Endpoints())
    })
}
```

Reply:
Your approach is much cleaner, so I adopted it. However, I combined it with a single scanner object to further reduce the number of lines.